### PR TITLE
Adding a draft of rank preference of using software on rhino/gizmo

### PR DESCRIPTION
**Pull request by laderast**
Hi Everyone,

I took a stab at trying to summarize the order in which users should try to run software on `rhino` / `gizmo`.

DaSL has taken a look at this internally, but we'd love any edits/opinions. I might be wrong in sections about best practices. In that case, feel free to suggest another route.

Ted

### CODE REVIEW

File: _dasldemos/computational_rank.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+Use Workflow Description Language (WDL) or Nextflow to orchestrate running containers on your data. The execution engine (such as miniWDL, Cromwell, or Nextflow) will handle loading the container as different tasks are run in your WDL script.
+
+There is a graphical user interface to running WDL scripts on your data on `gizmo` called [PROOF](https://proof.fredhutch.org/).
+
+### Containers Option 3
+
+Use Nextflow to run your software. Requires knowledge of Nextflow and Nextflow workflows.
+
+## 3. Best Practices When Using Conda
```
Comment (atombaby) [inline _dasldemos/computational_rank.md:109]: I've started a [page](https://github.com/FredHutch/wiki/pull/935/files) with some opinions on Python virtual environments... I'm not sure how that might integrate with these docs here, but we should make sure we're being consistent.

File: _dasldemos/computational_rank.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+---
+
+In this page, we talk about reproducible best practices to use software on the FH cluster (`rhino` / `gizmo`). If you're getting started on `rhino` / `gizmo`, this page is for you.
+
+## Learning Objectives
+
+After reading this page, you will be able to
+
+- **Describe** what the dependency nightmare is for software
+- **Utilize** software easymodules on `rhino`/`gizmo` and integrate them into your scripts
```
Comment (atombaby) [inline _dasldemos/computational_rank.md:14]: On terminology- "easymodule" isn't a term I've seen for these. There are "easyconfigs" used by "easybuild" which builds the software but what we're deploying is an "Lmod environment module", which is a bit cumbersome.  

Most of our [existing documentation](https://github.com/FredHutch/wiki/blob/2f86403d9e40deec702f69ad98d3bf7a5a7b467d/_scicomputing/compute_environments.md?plain=1#L17) refers to these as "environment modules"

File: _dasldemos/computational_rank.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+### In your script
+
+1. [The first line should be to load the module script](https://sciwiki.fredhutch.org/scicomputing/compute_environments/#best-practice-1): `source /app/lmod/lmod/init/profile`
+2. `module load` when about to use software in your script (use specific version numbers). For example: `module load SAMtools/0.1.20-foss-2018b` 
+3. As a dry run on the command line, `module load` and check to see that you can see the executable using `which` . For example: `which samtools` should return the path where this binary is installed (`/app/software/SAMtools/0.1.20-foss-2018b/bin/samtools`).
+4. Do you work in the script. For example: `samtools view -c myfile_bam > counts.txt`
+5. `module unload` your module or `module purge` all modules when done using them and starting a new step. Try not to load all modules all at once at beginning of script. For example: `module unload SAMtools/0.1.20-foss-2018b` 
+
+Here’s an example illustrating this:
```
Comment (atombaby) [inline _dasldemos/computational_rank.md:82]: I think an example is missing here... need one?

File: _dasldemos/computational_rank.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+### Containers Option 2
+
+Use Workflow Description Language (WDL) or Nextflow to orchestrate running containers on your data. The execution engine (such as miniWDL, Cromwell, or Nextflow) will handle loading the container as different tasks are run in your WDL script.
+
+There is a graphical user interface to running WDL scripts on your data on `gizmo` called [PROOF](https://proof.fredhutch.org/).
+
+### Containers Option 3
+
+Use Nextflow to run your software. Requires knowledge of Nextflow and Nextflow workflows.
```
Comment (atombaby) [inline _dasldemos/computational_rank.md:107]: Add links to nextflow docs (e.g. https://sciwiki.fredhutch.org/hdc/hdc_workflows/)

File: _dasldemos/computational_rank.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+2. Make sure that your software is available using `which` : `which samtools` . This should return a path to where your `samtools` binary is installed.
+3. Do your work in your script with the software.
+4. When you're done with that step, `conda deactivate` the `conda` environment. For example: `conda deactivate samtools_env` 
+
+## 4. Best Practices When Compiling from Source
+
+1. Double check to make sure that there isn’t a container or a precompiled binary available. I’m serious. This is going to take a while, and you want to make sure that you are using your time wisely. 
+2. Ask SciComp if it’s possible to make it into a module (they know a lot!)
+3. Before doing anything: look on Biostars or Stack Overflow to see if there’s compilation advice.
+4. Compile following the instructions. Be prepared to debug a number of steps.
```
Comment (atombaby) [inline _dasldemos/computational_rank.md:141]: The compilers on gizmo are very old... in most cases if someone _insists_ on compiling code we suggest loading a core environment module.  There are frequently other dependencies (BLAS is a big one) that are not available in the base OS on the rhino/gizmo nodes.  e.g. `ml spider foss/2023b`

Then they need to be certain to load that module before using the compiled binaries.

**Review by atombaby — COMMENTED** [review]
The biggest issue I see is with any Conda/virtual environment conflicts between this and the virtual environment doc I'd started.

File: _dasldemos/computational_rank.md (hunk 3 above)
Comment (laderast) [reply _dasldemos/computational_rank.md:82]: Yes, I'll make sure this is added.

File: _dasldemos/computational_rank.md (hunk 1 above)
Comment (laderast) [reply _dasldemos/computational_rank.md:129]: I've removed the installing the miniconda from script and moved it over to loading the Anaconda environment module. Hopefully this resolves the discrepancy

File: _dasldemos/computational_rank.md (hunk 2 above)
Comment (laderast) [reply _dasldemos/computational_rank.md]: Changed wording from Easymodules to Environment Modules

**Comment by laderast** [comment]
Could someone in @FredHutch/wiki-reviewers take a final look and merge? I've incorporated @atombaby 's suggestions.

**Review by atombaby — APPROVED** [review]

**Pull request merged**